logger.go

```diff
@@ -242,6 +245,64 @@ func (log *Logger) Fatal(msg string, fields ...Field) {
 	}
 }
 
+// InfoCtx logs a message at InfoLevel. The message includes any request context's fields
+// and any fields passed at the log site, as well as any fields accumulated on the logger.
+func (log *Logger) InfoCtx(ctx context.Context, msg string, fields ...Field) {
+	generatedFields := log.generateFields(ctx, fields...)
+	log.Info(msg, generatedFields...)
```

Comment:
To be the most optimal, I would do it differently. Change the others to call into this with Background or TODO context and check for level before generating the ctx fields.

Reply:
You're right!. I agree with you on checking for level. But I didn't understand call into with Background or TODO context. I'll revise the check level.

Reply:
I changed as you said. Thanks @bogdandrutu